### DISCUSSION TITLE
Master branch dont show in gh-pages

### DISCUSSION THREAD

**Discussion by ejgutierrez74** [community/community#22604] [Other Feature Feedback, Questions, & Ideas]
<p>Hello, ive followed an old tutorial about publishing in github pages.</p>
<p>The thing is that they recommend to create a gh-pages branch and to assure delete the master branch. So you are safe and the web published in <a href="http://github.io" rel="nofollow noopener">github.io</a>…</p>
<p>Recently ive read github documentation and it says that you can use either master or gh-branches…</p>
<p>The problem im facing is that many of my students can watch the web with a master branch ( no gh-pages created).</p>
<p>But one of my students didnt work… He gets a 404 error, althoug he has a index.html found…</p>
<p>You could think is some path error or mistake.</p>
<p>But he created a gh-pages and it worked as expected. So no error or linking related…</p>
<p>If he deleted the gh-pages again, it show the 404 error…</p>
<p>Perhaps a bug ?</p>
<p>Thanks</p>

**Comment by BaconBoi95** [comment]
<p>Can you show me a picture of his code</p>

**Comment by BaconBoi95** [comment]
<p>also show me a picture of his settings at the github pages section</p>

**Comment by BaconBoi95** [comment]
<p>merge gh-pages branch to master and set the source to master</p>

**Comment by ejgutierrez74** [comment]
<p>In this repository ( is public you can take a look whatever you want):</p>
<p>Repository : <a href="https://github.com/Werner365/RemoteDiariWerner" rel="nofollow noopener">https://github.com/Werner365/RemoteDiariWerner</a></p>
<p>Web ( gh-pages) :<a href="https://werner365.github.io/RemoteDiariWerner/" rel="nofollow noopener">https://werner365.github.io/RemoteDiariWerner/</a></p>
<p>The student couldnt use the master branch because it didnt work ( error 404), I get it to work creating a new branch gh-pages i then it worked. I dont know why if i only has the master branch it doesnt work.</p>
<p>Thanks</p>

**Comment by BaconBoi95** [comment]
<p>Ok, I will take a look</p>

**Comment by BaconBoi95** [comment]
<p>I will look at everything except the settings</p>
<p>i need admin privilages to see settings</p>

**Comment by BaconBoi95** [comment]
<p>First go to settings</p>
<p>Next scroll down to GitHub Pages</p>
<p>Next click on the Source dropdown</p>
<p>Now click Master</p>
<p>Now press Save</p>
<p>Finaly merge gh-pages into master</p>

**Comment by ejgutierrez74** [comment, marked as answer]
<p>Seems that the problem was that github pages takes some time to load. The student revert the changes and seems to work. Sorry for the inconvenience. If its any updates about this issue id report.</p>
<p>The objective of the question was not merging into one branch, but to guess why master branch wasnt published as readed in documentation. Seems that some orders ( specially change in css) take some time and others are almost live changes.</p>
<p>Thanks for your help.</p>
